README.md

```diff
@@ -0,0 +1,142 @@
+# Pactum
+## The HTTP-API specification sketchbook for pythonistas
+
+[![Circle CI](https://circleci.com/gh/olist/pactum.svg?style=svg)](https://circleci.com/gh/olist/pactum)
+
+```shell
+pip install pactum
+```
+
+With `Pactum` you can **specify** HTTP-APIs using pure python.
+
+Pactum is easy to use, easy to extend and easy to contribute:
+
+### Easy to use
+
+The only requirements to start writing an API specification with `pactum`
+is `pactum` package itself and some knowledge of python.
+
+```python
+import pactum
+
+class MyAPI(API):
+    name = 'My API'
+    versions = [...]
+```
+
+### Easy to extend
+
+Using the [visitor pattern](http://wiki.c2.com/?VisitorPattern) you can create
+exporters and extensions for any format or service you want.
+
+Take a look at [pactum/exporters/openapi.py](https://github.com/olist/pactum/blob/master/pactum/exporters/openapi.py).
+
+### Architecture
+
+Always keep [this diagram](https://github.com/olist/pactum/wiki/Architecture-Diagram) in mind when defining your APIs.
+
+
+### Tutorial
+
+Create a file called specs.py and start defining your API.
+
+You can define a `Resource` object for your API.
+
+```python
+from pactum import Action, API, Resource, Response, Version
+from pactum import fields, verbs
+
+class OrderResource(Resource):
+    fields = [
```

Comment:
why not use class attributes to define fields?

Reply:
I commented this because I faced all those problems with simple model last year as it had similar interface. But after a long time thinking and studying actual use cases I ended up changing to class attributes mostly because of one edge case: inheritance.

as simple model (version < 2) was getting more used some cases of inheritance started to appear and the subclasses got really ugly, hard to implement and all in all limited.

and it made me think: why models are defined as classes if we can't use inheritance properly? IMHO not being able to use OOP fully is a really drawback. at least that was the case for simple model, I'm not sure how this will work out for pactum.

with fields inheritance looks something like:

```python

class Order(Resource):
    fields = [
        fields.StringField(name='name'),
        fields.StringField(name='channel_slug'),
        ...  # imagine some other 10+ fields here
    ]


class SellerOrder(Order):
    fields = Order.fields + [fields.UUIDField(name='seller_id')]  # and we can't remove (easily) the channel_slug or other fields from the subclass without rewriting the fields attribute
```

maybe inheritance isn't an important feature for pactum, so using `fields` really might be the better approach as it really is easier to implement.

also, another lesson from simple-model: breaking retro compatibility and changing the interface from fields to class attributes is really hard on the user. as it requires a lot of "boring work" (probably this was harder on simple model, but still it's not going to be easy). 

therefore I recommend thinking hard on a """final""" interface on this before using it on a wide range of projects in order to "help" pactum users.